packages/ckeditor5-link/src/linkimageui.js

```diff
@@ -35,5 +40,82 @@ export default class LinkImageUI extends Plugin {
 	}
 
 	init() {
+		const editor = this.editor;
+		const viewDocument = editor.editing.view.document;
+
+		/**
+		 * The plugin button view.
+		 *
+		 * @member {module:ui/button/buttonview~Button}
+		 */
+		this.linkButtonView = null;
+
+		this.listenTo( viewDocument, 'click', ( evt, data ) => {
+			const hasLink = this._isImageLinked( editor.model.document.selection.getSelectedElement() );
+
+			if ( hasLink ) {
+				data.preventDefault();
+			}
+		} );
+
+		this._createToolbarLinkImageButton();
+	}
+
+	/**
+	 * Creates a LinkImageUI button view.
+	 * Clicking on the button shows a {@link module:link/linkui~LinkUI#_balloon} attached to the selection.
+	 * When image is already linked, the view shows {@link module:link/linkui~LinkUI#actionsView} or
+	 * {@link module:link/linkui~LinkUI#formView} if it's not.
+	 *
+	 * @private
+	 */
+	_createToolbarLinkImageButton() {
+		const editor = this.editor;
+		const t = editor.t;
+
+		editor.ui.componentFactory.add( 'linkImage', locale => {
+			const button = new ButtonView( locale );
+			const plugin = editor.plugins.get( 'LinkUI' );
+			const linkCommand = editor.commands.get( 'link' );
+
+			button.isEnabled = true;
+			button.label = t( 'Link image' );
+			button.icon = linkIcon;
+			button.keystroke = linkKeystroke;
+			button.tooltip = true;
+			button.isToggleable = true;
```

Comment:
```suggestion
			button.set( {
				isEnabled: true,
				label: t( 'Link image' ),
				icon: linkIcon,
				keystroke: linkKeystroke,
				tooltip: true,
				isToggleable: true
			} );
```

Let's keep that consistent (at least I've seen that more often recently).